client/web/src/repo/blame/useBlameHunks.ts

```diff
@@ -79,6 +104,82 @@ const fetchBlame = memoizeObservable(
+    author: {
+        Name: string
+        Email: string
+        Date: string
+    }
+    commit: {
+        parents: string[]
+        url: string
+    }
+    commitID: string
+    endLine: number
+    startLine: number
+    filename: string
+    message: string
+}
+
+const fetchBlameViaStreaming = memoizeObservable(
+    ({
+        repoName,
+        revision,
+        filePath,
+    }: {
+        repoName: string
+        revision: string
+        filePath: string
+    }): Observable<Omit<BlameHunk, 'displayInfo'>[] | undefined> =>
+        new Observable<Omit<BlameHunk, 'displayInfo'>[] | undefined>(subscriber => {
+            let assembledHunks: Omit<BlameHunk, 'displayInfo'>[] = []
+            const repoAndRevisionPath = `/${repoName}${revision ? `@${revision}` : ''}`
+            fetchEventSource(`/.api/blame${repoAndRevisionPath}/stream/${filePath}`, {
+                method: 'GET',
+                headers: {
+                    'X-Requested-With': 'Sourcegraph',
+                    'X-Sourcegraph-Should-Trace': new URLSearchParams(window.location.search).get('trace') || 'false',
+                },
+                onmessage(event) {
+                    if (event.event === 'hunk') {
+                        const rawHunks: RawStreamHunk[] = JSON.parse(event.data)
+                        for (const hunk of rawHunks) {
+                            assembledHunks.push({
+                                startLine: hunk.startLine,
+                                endLine: hunk.endLine,
+                                message: hunk.message,
+                                rev: hunk.commitID,
+                                author: {
+                                    date: hunk.author.Date,
+                                    person: {
+                                        email: hunk.author.Email,
+                                        displayName: hunk.author.Name,
+                                        user: null,
+                                    },
+                                },
+                                commit: {
+                                    url: hunk.commit.url,
+                                    parents: hunk.commit.parents ? hunk.commit.parents.map(oid => ({ oid })) : [],
+                                },
+                            })
+                        }
+                        subscriber.next(assembledHunks)
+                        assembledHunks = [...assembledHunks]
+                    }
+                },
+                onerror(event) {
+                    // eslint-disable-next-line no-console
+                    console.error(event)
+                },
+            }).then(
+                () => subscriber.complete(),
+                error => subscriber.error(error)
+            )
+            // Throttle the results to avoid re-rendering the blame sidebar for every hunk
+        }).pipe(throttleTime(1000, asyncScheduler, { leading: true, trailing: true })),
```

Comment:
```suggestion
        }).pipe(throttleTime(1000, undefined, { leading: true, trailing: true })),
```
nit: @philipp-spiess, `asyncScheduler` seems to be a [default value](https://rxjs.dev/api/index/function/throttleTime) so we can skip it ([usages](https://sourcegraph.com/search?q=context:%40taras-yemets/sg+throttletime%28&patternType=standard&sm=1))

Reply:
Done!